File: js/cm-api-v2/PostSnippet.js
// DocSection: cm_api_v2_post_snippet
// Tip: Find more about JS/TS SDKs at https://docs.kontent.ai/javascript
// Using ES6 syntax
import { ManagementClient } from '@kentico/kontent-management';

const client = new ManagementClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

client.addContentTypeSnippet()
   .withData(builder => {
       return {
          name: "metadata",
          codename: "my_metadata",
          external_id: "snippet-item-123",
          elements: [
             builder.textElement({
               name: "Meta title",
               codename: "my_metadata__meta_title",
               guidelines: "Length: 30–60 characters",
               type: ElementModels.ElementType.text,
               external_id: "meta_title"
             }),
             builder.textElement({
               name: "Meta description",
               codename: "my_metadata__meta_description",
               guidelines: "Length: 70-150 characters",
               type: ElementModels.ElementType.text,
               external_id: "meta_title"
             })
          ]
       };
   })

  .toObservable()
  .subscribe((response) => {
    console.log(response);
  },
  (error) => {
    console.log(error);
  });
// EndDocSection

File: net/management-api-v2/PostSnippet.cs
// DocSection: cm_api_v2_post_snippet
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kentico.Kontent.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_PROJECT_ID>"
});

var response = await client.CreateContentTypeSnippetAsync(new ContentTypeSnippetCreateModel
{
    Name = "metadata",
    Codename = "my_metadata",
    ExternalId = "snippet-item-123",
    Elements = new ElementMetadataBase[]
    {
        new TextElementMetadataModel
        {
            Name = "Meta title",
            Codename = "my_metadata__meta_title",
            Guidelines = "Length: 30-60 characters",
            ExternalId = "meta_title",
        },
        new TextElementMetadataModel
        {
            Name = "Meta description",
            Codename = "my_metadata__meta_description",
            Guidelines = "Length: 70-11500 characters",
            ExternalId = "meta_description",
        }
    }
});
// EndDocSection